"make dev-setup" returns "Unsupported Kubernetes Environment"

**What happened**:
`make dev-setup` returns `Unsupported Kubernetes Environment`. With the latest minikube versions the minikube node name is no longer `minikube`.

**How to reproduce it (as minimally and precisely as possible)**:
```
$ minikube version
minikube version: v1.8.2
commit: eb13446e786c9ef70cb0a9f85a633194e62396a1

$ k get node
NAME   STATUS   ROLES    AGE     VERSION
m01    Ready    master   2m49s   v1.18.0-rc.1

$ make dev-setup
Unsupported Kubernetes Environment:
make: *** [dev-setup] Error 1
```

- Gardener version: master
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Others:

--- a/docs/development/local_setup.md
+++ b/docs/development/local_setup.md
@@ -177,7 +177,7 @@ Alternatively, spin up a cluster with minikube with this command:
 
 ```bash
 minikube start --embed-certs #  `--embed-certs` can be omitted if minikube has already been set to create self-contained kubeconfig files.
-😄  minikube v1.5.2 on darwin (amd64)
+😄  minikube v1.8.2 on Darwin 10.15.3
 🔥  Creating virtualbox VM (CPUs=2, Memory=2048MB, Disk=20000MB) ...
 [...]
 🏄  Done! Thank you for using minikube!

--- a/hack/common
+++ b/hack/common
@@ -42,7 +42,8 @@ k8s_nodename() {
 
 k8s_env() {
     node_name=$(k8s_nodename)
-    if [[ "$node_name" == "$MINIKUBE" ]]; then
+    # for minikube >= v1.8.0 the default node name is "m01", for minikube < v1.8.0 - "minikube"
+    if [[ "$node_name" == "$MINIKUBE" || "$node_name" == "m01" ]]; then
         echo "$MINIKUBE"
         return
     fi

